src/common/models/project.go

```diff
@@ -146,6 +210,36 @@ type ProjectQueryParam struct {
 	ProjectIDs []int64      // project ID list
 }
 
+// ToQuery returns q.Query from param
+func (param *ProjectQueryParam) ToQuery() *q.Query {
+	kw := q.KeyWords{}
+	if param.Name != "" {
+		kw["name"] = q.FuzzyMatchValue{Value: param.Name}
+	}
+	if param.Owner != "" {
+		kw["owner"] = param.Owner
+	}
+	if param.Public != nil {
+		kw["public"] = *param.Public
```

Comment:
*param.Public？

Reply:
Yes, convert the pointer to the value because the `QueryPublic` method of `Project` model only accepts `string` and `bool` value to do filter.